### ISSUE TITLE
Add/Update realmObject in RealmList

### ISSUE THREAD

**Issue by AironVarun** [realm/realm-java#4479]
I have a realmList, say studentsList within a realmObject, say class. I want to request a list of students for a given class and update the same within studentList. Server returned studentList may contain old as well as new students. So I want my realmList to update information for old students along with addition of new students if any.

What I did is ,

1. Initiated a loop with size == serverReturnedStudentList.size()
2. Created student within loop and added in realm (using realm.copyToRealmOrUpdate(student))
3. Received managed student object from step 2.
4. Fetch student List for class X and checked if the studentList contains that managedObject or not.
5. If not, added student to studentList else did nothing.
6. It is working as expected but I know this is not the right approach. If I skip that "contains" check (@step4) and directly add managed object to realmList, multiple entries starts displaying in realmList.

What would be the best approach to handle this scenario?

Another issue is, I have 'n' number of class and 'x' sections under each class. I am making n.x requests in parallel and when I try to start asyncExecutions n.x times, I got pool size limit exception ( which is obvious) and I can't run this on UI thread as my UI becomes unresponsive for 3-5 seconds(while the loop for each studentList is getting executed with executeTransaction for each student)

Any help/suggestion would be highly appreciated.

Thanks in Advance. Cheers!

**Comment by zaki50** [comment]
@AironVarun We've started a policy that we use github issues only for bugs or feature request.
Please ask on StackOverflow which we also monitor and answer.

And could you create separate questions for different topics.

Thanks!

**Comment by Zhuinden** [comment]
http://stackoverflow.com/questions/43349261/add-update-realmobject-in-realmlist

**Comment by zaki50** [comment]
@Zhuinden thanks.

**Comment by Zhuinden** [comment]
I kinda got stumped though about it

**Comment by AironVarun** [comment]
@zaki50 I have already asked the same on StackOverFlow and @Zhuinden thanks for sharing the stackoverflow link here.

Still looking for some better approach.

**Comment by Zhuinden** [comment]
If you share the code instead of describing the code, I might be able to figure something out.
